splunklogger.js

```diff
@@ -107,6 +136,58 @@ var defaultRequestOptions = {
 };
 
 /**
+ * Disables the interval timer set by <code>this._enableTimer()</code>.
+ *
+ * param {Number} interval - The batch interval.
+ * @private
+ */
+SplunkLogger.prototype._disableTimer = function() {
+    if (this._timerID) {
+        clearInterval(this._timerID);
+        this._timerDuration = 0;
+        this._timerID = null;
+    }
+};
+
+/**
+ * Configures an interval timer to flush any events in
+ * <code>this.contextQueue</code> at the specified interval.
+ *
+ * param {Number} interval - The batch interval.
+ * @private
+ */
+SplunkLogger.prototype._enableTimer = function(interval) {
+    // Only enable the timer if possible
+    if (typeof interval === "number") {
+        if (interval <= 0) {
+            throw new Error("Batch interval must be a positive number, found: " + interval);
+        }
+        else {
+            if (this._timerID) {
+                this._disableTimer();
+            }
+            
+            // If batch interval is changed, update the config property
+            if (this.config) {
+                this.config.batchInterval = interval;
+            }
+
+            this._timerDuration = interval;
```

Comment:
_timerInterval looks better to me. Feel free to ignore this comment if you still feel that "duration" is prefered (I'm absolutely sure your English is much-much better than mine so that's up to you).